src/cpr_data_access/pipeline_general_models.py

```diff
@@ -0,0 +1,118 @@
+from datetime import datetime
+from enum import Enum
+from typing import Mapping, Any, List, Optional, Sequence, Union
+
+from pydantic import BaseModel, Extra, root_validator
+
+Json = dict[str, Any]
+
+CONTENT_TYPE_HTML = "text/html"
+CONTENT_TYPE_DOCX = (
+    "application/vnd.openxmlformats-officedocument.wordprocessingml.document"
+)
+CONTENT_TYPE_PDF = "application/pdf"
+
+
+class DocumentMetadata(BaseModel, extra=Extra.allow):
+    """Metadata about a document."""
+
+    metadata: Optional[Json] = {}
+    publication_ts: datetime
+    date: Optional[str] = None  # Set on import by a validator
+    geography: str
+    category: str
+    source: str
+    type: str
+
+    @root_validator
+    def convert_publication_ts_to_date(cls, values):
```

Comment:
Merge this with BackendDocument by bringing down the method  and the date field.